### DISCUSSION TITLE
Dotfiles: how can I add aliases/etc to the default shell?

### DISCUSSION THREAD

**Discussion by julien-c** [community/community#4172] [Codespaces] [answered]
### What I tried:
- adding my stuff to a `.profile` or `.bash_profile` in my `dotfiles` repo (but it looks like the Codespaces shells are non-login shells)
- I also tried adding to .bashrc from `install.sh` (https://github.com/julien-c/dotfiles/commit/ad01ad8ec35166a21eaf8da1086aab3628882982) but didn't work either. 

I'm sure I'm missing something obvious here! Any idea?

cc @osanseviero

**Comment by dynamicwebpaige** [comment]
Tagging in @Chuxel to help - your `.bash_profile` might need to be added to the [`devcontainer.json`](https://code.visualstudio.com/docs/remote/devcontainerjson-reference), particularly `userEnvProbe`. And hi, @julien-c! 🤗

**Reply by julien-c** [reply]
Hi Paige! 👋

**Comment by Chuxel** [comment, marked as answer]
Hmmm. `~/.bashrc` should work. Where are you not seeing it take effect?  Is this a custom image (e.g. is the terminal actually sh not bash)?

If you want VS Code itself to inherit from `~/.bashrc` or `~/.profile`, there is a `devcontainer.json` property called `userEnvProbe`. You can set this to `interactiveLoginShell` to get anything set in places like `/etc/profile` or `~/.profile` to kick in. This setting comes with a startup perf hit which is why it's not currently the default - but I'm hoping we can reduce it to the point we switch to it.

For the integrated terminal, the default profile on Linux does indeed appear to be interactive non-login shell even outside of Codespaces - I checked and the same thing is true locally on Linux and in WSL. (I assume this is intentional @Tyriar ?)

You can change this for VS Code in your local settings if you have settings sync enabled or through the `settings` property in `devcontainer.json` using terminal profiles. [See here](https://code.visualstudio.com/docs/editor/integrated-terminal#_configuring-profiles).  You're probably used to `args` being `["-li"]`.  These settings are OS specific, so you can change the default for Linux without affecting macOS.

**Reply by Tyriar** [reply]
> I checked and the same thing is true locally on Linux and in WSL. (I assume this is intentional @Tyriar ?)

I think this was intentional at the time, when I brought it up recently there was a lot of hesitation in changing it as it will likely have a lot of knock on effects.

**Reply by julien-c** [reply]
Sorry i'm super late to this thread, but what do you mean exactly by "If you want VS Code itself to inherit from `~/.bashrc` or `~/.profile`"? This would be mostly to source env variables, no? or would there be any other use?

(for context, yeah my initial question was about the integrated terminal)

**Reply by julien-c** [reply]
Also for future readers, it does look like `loginInteractiveShell` is enabled by default for `userEnvProbe` now
